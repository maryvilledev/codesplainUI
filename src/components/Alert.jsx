import React, { PropTypes } from 'react';
import {
  Dialog,
  FlatButton,
} from 'material-ui';

/*
<Alert /> renders as a <Dialog /> with an OK button. It displays the text
specified in the 'text' prop, and invokes the 'onClose' prop when it is closed.
Setting the 'isError' prop to true causes its background to be rendered as a
crimson red.
*/
const Alert = ({ onClose, text }) => (
    <Dialog
      actions={
        <FlatButton
          label="OK"
          primary={true}
          onTouchTap={onClose}
        />
      }
      modal={false}
      open={true}
      onRequestClose={onClose}
    >
      {text}
    </Dialog>
);

Alert.propTypes = {
  onClose: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
};

export default Alert;
