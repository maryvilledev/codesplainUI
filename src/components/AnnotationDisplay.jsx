import React, { PropTypes } from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import IconButton from 'material-ui/IconButton';
import Previous from 'material-ui/svg-icons/navigation/arrow-back';
import Next from 'material-ui/svg-icons/navigation/arrow-forward';

import MarkdownDisplayer from './MarkdownDisplayer';

const styles = {
  actionRow: {
    alignItems: 'center',
    display: 'flex',
    flexFlow: 'row wrap',
    justifyContent: 'space-between',
  },
  annotationViewButtons: {
    display: 'flex',
    flex: '0 1 auto',
    justifyContent: 'flex-end',
  },
  cancelButton: {
    marginRight: '0.2rem',
  },
};

const AnnotationDisplay = (props) => {
  const {
    annotation,
    closeAnnotation,
    editAnnotation,
    getNextAnnotation,
    getPreviousAnnotation,
    hasNextAnnotation,
    hasPrevAnnotation,
  } = props;

  return (
    <div>
      <MarkdownDisplayer annotation={annotation} />
      <div style={styles.actionRow}>
        <div style={styles.actionButtons}>
          <RaisedButton
            label="Close"
            onTouchTap={closeAnnotation}
            secondary
            style={styles.cancelButton}
          />
          <RaisedButton
            label="Edit"
            onTouchTap={editAnnotation}
            primary
          />
        </div>
        <div style={styles.annotationViewButtons}>
          <IconButton
            disabled={hasPrevAnnotation}
            id="previous-annotation"
            onTouchTap={getPreviousAnnotation}
            tooltip="Previous Annotation"
            touch
          >
            <Previous />
          </IconButton>
          <IconButton
            disabled={hasNextAnnotation}
            id="next-annotation"
            onTouchTap={getNextAnnotation}
            tooltip="Next Annotation"
            touch
          >
            <Next />
          </IconButton>
        </div>
      </div>
    </div>
  );
};

AnnotationDisplay.propTypes = {
  annotation: PropTypes.string.isRequired,
  closeAnnotation: PropTypes.func.isRequired,
  editAnnotation: PropTypes.func.isRequired,
  getNextAnnotation: PropTypes.func.isRequired,
  getPreviousAnnotation: PropTypes.func.isRequired,
  hasNextAnnotation: PropTypes.bool.isRequired,
  hasPrevAnnotation: PropTypes.bool.isRequired,
};

export default AnnotationDisplay;
