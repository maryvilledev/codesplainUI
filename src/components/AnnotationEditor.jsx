import React, { PropTypes } from 'react';
import { Tab, Tabs } from 'material-ui/Tabs';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';

import MarkdownDisplayer from './MarkdownDisplayer';
import markdownLogo from '../../res/markdown-logo.svg';

const styles = {
  bottomContainer: {
    display: 'flex',
    flexFlow: 'row wrap',
    alignItems: 'center',
    marginTop: '0.5rem',
  },
  button: {
    flexFlow: 1,
    flexShrink: 1,
    flexBasis: 'auto',
  },
  cancelButton: {
    flexFlow: 1,
    flexShrink: 1,
    flexBasis: 'auto',
    marginRight: '0.2rem',
  },
  hintText: {
    top: '1rem',
  },
  markdownHintText: {
    color: '#d3d3d3',
  },
  markdownIndicator: {
    paddingLeft: '1rem',
    flexFlow: 1,
    flexShrink: 1,
    flexBasis: 'auto',
  },
  markdownLogo: {
    width: '2rem',
    height: '1rem',
  },
};

class AnnotationEditor extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      annotation: this.props.annotation,
    };
    this.clearAnnotation = this.clearAnnotation.bind(this);
    this.onAnnotationChange = this.onAnnotationChange.bind(this);
    this.saveAnnotation = this.saveAnnotation.bind(this);
  }

  onAnnotationChange(ev, annotation) {
    this.setState({
      annotation,
    });
  }

  clearAnnotation() {
    this.onAnnotationChange(null, '');
    this.props.closeAnnotation();
  }

  saveAnnotation() {
    const { annotation } = this.state;
    this.props.saveAnnotation(annotation);
  }

  render() {
    const { annotation } = this.state;
    return (
      <div>
        <Tabs>
          <Tab
            label="Write"
            onActive={() => { this.textField.focus(); }}
          >
            <TextField
              autoFocus
              fullWidth
              hintStyle={styles.hintText}
              hintText="Enter your annotation here"
              multiLine
              name="annotationEditor"
              onChange={this.onAnnotationChange}
              ref={(textField) => { this.textField = textField; }}
              rows={4}
              value={annotation}
            />
          </Tab>
          <Tab label="Preview">
            <MarkdownDisplayer annotation={annotation} />
          </Tab>
        </Tabs>
        <div style={styles.bottomContainer}>
          <RaisedButton
            label="Cancel"
            onTouchTap={this.clearAnnotation}
            secondary
            style={styles.cancelButton}
          />
          <RaisedButton
            disabled={!annotation}
            label="Save"
            onTouchTap={this.saveAnnotation}
            primary
            style={styles.button}
          />
          <a
            href="http://commonmark.org/help"
            rel="noopener noreferrer"
            style={styles.markdownIndicator}
            target="_blank"
          >
            <img
              alt="Markdown Logo"
              src={markdownLogo}
              style={styles.markdownLogo}
            />
            <span style={styles.markdownHintText}>
              Styling with Markdown is supported
            </span>
          </a>
        </div>
      </div>
    );
  }
}

AnnotationEditor.defaultProps = {
  annotation: '',
};

AnnotationEditor.propTypes = {
  annotation: PropTypes.string,
  closeAnnotation: PropTypes.func.isRequired,
  saveAnnotation: PropTypes.func.isRequired,
};

export default AnnotationEditor;
