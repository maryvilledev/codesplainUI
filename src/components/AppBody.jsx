import React from 'react';
import { browserHistory } from 'react-router'
import { Card, CardText } from 'material-ui/Card';
import LanguageSelector from './LanguageSelector';
import SnippetArea from './SnippetArea';
import TokenSelector from './TokenSelector';
import TokenInfoPanel from './TokenInfoPanel';
import { parsePython3 } from '../parsers/python3';
import { highlight }  from '../util/highlight.js';
import axios from 'axios'

const languages = [
  { text: 'Go' , value: 'go' },
  { text: 'Python 3' , value: 'python3' },
];

const parsers = {
  python3: parsePython3,
}

const tokenTypes = [
  { text: "Function" },
  { text: "For loop" },
  { text: "If statement" },
  { text: "Variable" },
  { text: "Return statement" },
];

// Keeps track of how long until we're ready to parse another AST
let parseReady = true;

class AppBody extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      annotationDisplay: 'none',
      annotationDisplayProps: {},
      annotations: {  },
      AST: [],
      displayedAnnotation: null,
      filters: {},
      readOnly: false,
      selectedLanguage: 'python3',
      snippet: '',
      snippetEditorMode: '',
      snippetTitle: '',
    };
    this.closeAnnotation = this.closeAnnotation.bind(this);
    this.handleSelect = this.handleSelect.bind(this);
    this.onFiltersChanged = this.onFiltersChanged.bind(this);
    this.onGutterClick = this.onGutterClick.bind(this);
    this.onSaveState = this.onSaveState.bind(this);
    this.onSnippetChanged = this.onSnippetChanged.bind(this);
    this.onSnippetTitleChanged = this.onSnippetTitleChanged.bind(this);
    this.saveAnnotation = this.saveAnnotation.bind(this);
    this.switchReadOnlyMode = this.switchReadOnlyMode.bind(this);
  }

  onGutterClick(codeMirrorInstance, lineNumber) {
    if (this.state.annotations[String(lineNumber)] === undefined) {
      this.setState({
        annotationDisplay: 'create',
        annotationDisplayProps: {
          closeAnnotation: this.closeAnnotation,
          lineNumber,
          lineText: codeMirrorInstance.getLine(lineNumber),
          saveAnnotation: this.saveAnnotation,
          snippetLanguage: this.state.selectedLanguage,
        },
      });
      return;
    }
    this.setState({
      annotationDisplay: 'display',
      annotationDisplayProps: {
        closeAnnotation: this.closeAnnotation,
        lineNumber,
        lineText: codeMirrorInstance.getLine(lineNumber),
        snippetLanguage: this.state.selectedLanguage,
        text: this.state.annotations[String(lineNumber)],
      }
    })
  }

  closeAnnotation() {
    this.setState({
      annotationDisplay: 'none',
      annotationDisplayProps: {},
    });
  }

  handleSelect(ev, index, value) {
    this.setState({
      selectedLanguage: value,
    });
  }

  switchReadOnlyMode() {
    if (this.state.readOnly) {
      console.log("Switching back to editing mode not supported yet");
      return;
    }
    this.setState({
      readOnly: !(this.state.readOnly),
    });
  }

  // Callback to be invoked when user changes snippet title
  onSnippetTitleChanged(ev) {
    const title = ev.target.value;
    this.setState({ snippetTitle: title });
  }

  onSaveState() {
    const {snippet, snippetTitle, annotations, AST, filters} = this.state;
    const obj = {snippet, snippetTitle, annotations, AST, filters};
    const stateString = JSON.stringify(obj);
    axios.post('/api/snippets/', { json: stateString })
      .then(res => {
        const id = res.data.id;
        this.setState({ id: id });
        browserHistory.push(`/${id}`);
      })
      .catch(err => {
        console.error(err);
      })
  }

  componentDidMount() {
    const { id } = this.props.params;
    if (!id) return;
    axios.get(`/api/snippets/${id}`)
      .then(res => {
        const stateString = res.data.json;
        const obj = JSON.parse(stateString);
        this.setState(obj);
      })
      .catch(err => {
        // Bad URL, redirect
        browserHistory.push('/');
      });
  }

  // Callback to be invoked when user edits the code snippet
  onSnippetChanged(snippet, codeMirrorRef) {
    this.setState({ snippet });
    // Make sure a language is selected
    const currentLang = this.state.selectedLanguage;
    if (!currentLang) {
      console.warn('Cannot parse snippet. No language selected.');
      return;
    }

    // Map to selected language to its parser, if one exists
    const parser = parsers[currentLang];
    if (parser === undefined) {
      console.warn(`No parser available for the ${currentLang} language`);
      return;
    }

    // Generate an AST for the current state of the code snippet, then
    // highlight tokens in snippet and update state
    if(parseReady) {
      parseReady = false;
      setTimeout(() => parseReady = true, 1000);
      new Promise((resolve) => resolve(parser(snippet)))
      .then((AST) => {
        highlight(snippet, AST, codeMirrorRef);
        this.setState({ AST: AST });
      });
    }
  }

  onFiltersChanged(filters) {
    this.setState({ filters });
  }

  saveAnnotation(lineNumber, annotation) {
    const annotations = this.state.annotations;
    this.setState({
      annotations: {
        ...annotations,
        [lineNumber]: annotation,
      },
    });
  }

  render() {
    return (
      <div className="container-fluid">
        <div className="row">
          <div className="col-md-2">
            <Card>
              <CardText>
                <LanguageSelector
                  languages={languages}
                  onChange={this.handleSelect}
                  selected={this.state.selectedLanguage}
                />
                <TokenSelector
                  onChange={this.onFiltersChanged}
                  tokenTypes={tokenTypes}
                />
              </CardText>
            </Card>
          </div>
          <div className="col-md-6">
            <SnippetArea
              annotatedLines={Object.keys(this.state.annotations)}
              contents={this.state.snippet}
              onGutterClick={this.onGutterClick}
              onSaveClick={this.onSaveState}
              onSnippetChanged={this.onSnippetChanged}
              onTitleChanged={this.onSnippetTitleChanged}
              readOnly={this.state.readOnly}
              snippetLanguage={this.state.selectedLanguage}
              switchReadOnlyMode={this.switchReadOnlyMode}
              title={this.state.snippetTitle}
            />
          </div>
          <div className="col-md-4">
            <TokenInfoPanel
              displayProps={this.state.annotationDisplayProps}
              displayStatus={this.state.annotationDisplay}
            />
          </div>
        </div>
      </div>
    );
  }
}

export default AppBody;
