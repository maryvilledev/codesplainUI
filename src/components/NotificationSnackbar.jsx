import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import Snackbar from 'material-ui/Snackbar';

import { closeNotification } from '../actions/notifications';

export const NotificationSnackbar = (props) => {
  const {
    onRequestClose,
    displaying,
    notification,
  } = props;

  return (
    <Snackbar
      autoHideDuration={5000}
      message={notification}
      onRequestClose={onRequestClose}
      open={displaying}
    />
  );
};

NotificationSnackbar.propTypes = {
  displaying: PropTypes.bool.isRequired,
  notification: PropTypes.string,
  onRequestClose: PropTypes.func.isRequired,
};

NotificationSnackbar.defaultProps = {
  notification: '',
};

const mapDispatchToProps = dispatch => ({
  onRequestClose: () => { dispatch(closeNotification()); },
});

const mapStateToProps = (state) => {
  const {
    notifications: {
      displaying,
      notifications,
    },
  } = state;
  return {
    displaying,
    notification: notifications[0],
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(NotificationSnackbar);
