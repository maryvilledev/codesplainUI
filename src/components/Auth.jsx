import React from 'react'
import axios from 'axios'
import cookie from 'react-cookie'
import Loading from './Loading'
import Alert from './Alert'

/*
<Auth /> is the component that is rendered for the '{{url}}/auth' endpoint (which
is the endpoint the user is sent to after signing in to GitHub). It initially
renders as a <CircularProgress /> and sends a POST request to the express backend
to complete the OAuth authentication with GitHub. Upon completion of this, the
user is redirected to the URL they attempted to login from initially.
*/
class Auth extends React.Component {
  constructor(props) {
    super(props);
    this.state = { waiting: true };
    this.redirectUser = this.redirectUser.bind(this);
  }

  componentDidMount() {
    // Extract the code provided by GitHub from the redirect query string
    const { code } = this.props.location.query

    // Post it to the express backend to be verified by GitHub as authentic
    axios.post('/api/auth', { code })
      .then(res => {
        // Code was accepted, so extract and save the token from the response
        const { token } = res.data;
        cookie.save('token', token, { path: '/' });

        // Return Promise to get the user's basic info
        return axios.get('https://api.github.com/user', {
          headers: {
            Accept: 'application/json',
            Authorization: `token ${token}`,
          }
        })
      }, err => {
        // If this fails, we need to make sure the error dialog shows
        console.error(err);
        this.setState({ waiting: false, error: true });
      })
      .then(res => {
        const { avatar_url } = res.data; // Can pull lots of other stuff
                                         // out of res.data if needed
        cookie.save('userAvatarURL', avatar_url, { path: '/' });
        this.setState({ waiting: false });
      }, err => {
        // Failed to pull in user info, but that's fine. Log and continue.
        console.log(err);
        this.setState({ waiting: false });
      })
      .catch(err => console.log(err))
  }

  redirectUser() {
    // Load the URL to redirect user to, default to the
    // home page if non exists
    const signInRedirect = cookie.load('signInRedirect');
    if (signInRedirect) {
      window.location = signInRedirect;
    } else {
      window.location = '/';
    }
  }

  render() {
    // Conditionally render a <CircularProgress /> or redirect user, depending
    // on whether the backend has responded yet
    if (this.state.waiting) {
      return <Loading text="Logging in..." />
    } else {
      // We're done waiting, now render an alert if
      // the login attempt failed
      if (this.state.error) {
        return (
          <Alert
            text="Failed to login with GitHub, sorry."
            onClose={this.redirectUser}
          />
        );
      } else {
        // Otherwise, it succeeded, so just redirect the user
        this.redirectUser();
        return null;
      }
    }
  }
}

export default Auth
