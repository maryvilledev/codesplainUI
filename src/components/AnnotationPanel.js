import React, { PropTypes } from 'react';

import AnnotationDisplay from './AnnotationDisplay';
import AnnotationEditor from './AnnotationEditor';
import LineSnippet from './LineSnippet';

class AnnotationPanel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isEditing: props.annotation.length === 0,
    };
    this.handleSaveAnnotation = this.handleSaveAnnotation.bind(this);
    this.toggleEditState = this.toggleEditState.bind(this);
  }

  toggleEditState() {
    this.setState({
      isEditing: !this.state.isEditing,
    });
  }

  handleSaveAnnotation(annotation) {
    this.props.saveAnnotation(annotation);
    this.toggleEditState();
  }

  render() {
    const {
      annotation,
      closeAnnotation,
      snippetInformation: {
        lineNumber,
        lineText,
      }
    } = this.props;
    const { isEditing } = this.state;
    const shouldRenderEditor = annotation.length === 0 || isEditing;
    return (
      <div>
        <LineSnippet
          lineNumber={lineNumber + 1}
          value={lineText}
        />
        { shouldRenderEditor ?
          <AnnotationEditor
            annotation={annotation}
            closeAnnotation={closeAnnotation}
            saveAnnotation={this.handleSaveAnnotation}
          /> :
          <AnnotationDisplay
            annotation={annotation}
            closeAnnotation={closeAnnotation}
            editAnnotation={this.toggleEditState}
          />
        }
      </div>
    );
  }

}

AnnotationPanel.propTypes = {
  annotation: PropTypes.string.isRequired,
  saveAnnotation: PropTypes.func.isRequired,
  closeAnnotation: PropTypes.func.isRequired,
  snippetInformation: PropTypes.shape({
    lineNumber: PropTypes.number.isRequired,
    lineText: PropTypes.string.isRequired,
  }).isRequired,
};

export default AnnotationPanel;
