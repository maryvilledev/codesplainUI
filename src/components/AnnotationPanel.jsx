import React, { PropTypes } from 'react';

import AnnotationDisplay from './AnnotationDisplay';
import AnnotationEditor from './AnnotationEditor';
import LineSnippet from './LineSnippet';

const style = {
  container: {
    padding: '3%',
  },
};

class AnnotationPanel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isEditing: props.annotation.length === 0,
    };
    this.handleSaveAnnotation = this.handleSaveAnnotation.bind(this);
    this.toggleEditState = this.toggleEditState.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    const {
      lineAnnotated: {
        lineNumber,
      },
    } = this.props;
    if (lineNumber !== nextProps.lineAnnotated.lineNumber) {
      this.setState({
        isEditing: nextProps.annotation.length === 0,
      });
    }
  }

  handleSaveAnnotation(annotation) {
    this.props.saveAnnotation(annotation);
    this.toggleEditState();
  }

  toggleEditState() {
    this.setState({
      isEditing: !this.state.isEditing,
    });
  }

  render() {
    const {
      annotation,
      closeAnnotation,
      getNextAnnotation,
      getPreviousAnnotation,
      hasNextAnnotation,
      hasPrevAnnotation,
      lineAnnotated: {
        lineNumber,
        lineText,
      },
    } = this.props;
    const { isEditing } = this.state;
    return (
      <div style={style.container}>
        <LineSnippet
          lineNumber={lineNumber + 1}
          value={lineText}
        />
        { isEditing ?
          <AnnotationEditor
            annotation={annotation}
            closeAnnotation={closeAnnotation}
            saveAnnotation={this.handleSaveAnnotation}
          /> :
          <AnnotationDisplay
            annotation={annotation}
            closeAnnotation={closeAnnotation}
            editAnnotation={this.toggleEditState}
            getNextAnnotation={getNextAnnotation}
            getPreviousAnnotation={getPreviousAnnotation}
            hasPrevAnnotation={hasPrevAnnotation}
            hasNextAnnotation={hasNextAnnotation}
          />
        }
      </div>
    );
  }
}

AnnotationPanel.propTypes = {
  annotation: PropTypes.string.isRequired,
  closeAnnotation: PropTypes.func.isRequired,
  saveAnnotation: PropTypes.func.isRequired,
  getNextAnnotation: PropTypes.func.isRequired,
  getPreviousAnnotation: PropTypes.func.isRequired,
  hasNextAnnotation: PropTypes.bool.isRequired,
  hasPrevAnnotation: PropTypes.bool.isRequired,
  lineAnnotated: PropTypes.shape({
    lineNumber: PropTypes.number,
    lineText: PropTypes.string,
  }).isRequired,
};

export default AnnotationPanel;
