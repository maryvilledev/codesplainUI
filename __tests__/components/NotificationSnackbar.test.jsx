import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import { NotificationSnackbar, mapDispatchToProps } from '../../src/components/NotificationSnackbar';

const defaultProps = {
  displaying: false,
  notification: 'Grasssssss tastes baddd',
  onRequestClose: jest.fn(),
};

describe('<NotificationSnackbar />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  describe('snapshots', () => {
    it('matches when it is displayed', () => {
      const wrapper = shallowWithContext(
        <NotificationSnackbar
          {...defaultProps}
          displaying
        />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('matches when it is not displayed', () => {
      const wrapper = shallowWithContext(
        <NotificationSnackbar
          {...defaultProps}
        />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    describe('mapDispatchToProps', () => {
      it('matches snapshot', () => {
        expect(mapDispatchToProps).toMatchSnapshot();
      });
    });
  });
});
