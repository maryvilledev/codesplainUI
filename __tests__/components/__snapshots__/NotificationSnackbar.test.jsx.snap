exports[`<NotificationSnackbar /> snapshots mapDispatchToProps matches snapshot 1`] = `undefined`;

exports[`<NotificationSnackbar /> snapshots matches when it is displayed 1`] = `
<Snackbar
  autoHideDuration={5000}
  message="Grasssssss tastes baddd"
  onRequestClose={[Function]}
  open={true} />
`;

exports[`<NotificationSnackbar /> snapshots matches when it is not displayed 1`] = `
<Snackbar
  autoHideDuration={5000}
  message="Grasssssss tastes baddd"
  onRequestClose={[Function]}
  open={false} />
`;
